x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -93,89 +49,43 @@ export const signalRulesAlertType = ({
+        services,
+        ruleStatusSavedObjects,
+      });
+
+      const {
+        name,
+        tags,
+        createdAt,
+        createdBy,
+        updatedBy,
+        enabled,
+        schedule: { interval },
+      } = savedObject.attributes;
 
-      const createdBy = savedObject.attributes.createdBy;
-      const createdAt = savedObject.attributes.createdAt;
-      const updatedBy = savedObject.attributes.updatedBy;
       const updatedAt = savedObject.updated_at ?? '';
-      const interval = savedObject.attributes.schedule.interval;
-      const enabled = savedObject.attributes.enabled;
-      const gap = getGapBetweenRuns({
-        previousStartedAt: previousStartedAt != null ? moment(previousStartedAt) : null, // TODO: Remove this once previousStartedAt is no longer a string
-        interval,
-        from,
-        to,
-      });
-      if (gap != null && gap.asMilliseconds() > 0) {
-        logger.warn(
-          `Signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}" has a time gap of ${gap.humanize()} (${gap.asMilliseconds()}ms), and could be missing signals within that time. Consider increasing your look behind time or adding more Kibana instances.`
-        );
-        // write a failure status whenever we have a time gap
-        // this is a temporary solution until general activity
-        // monitoring is developed as a feature
-        const gapDate = new Date().toISOString();
-        await services.savedObjectsClient.create(ruleStatusSavedObjectType, {
-          alertId,
-          statusDate: gapDate,
-          status: 'failed',
-          lastFailureAt: gapDate,
-          lastSuccessAt: currentStatusSavedObject.attributes.lastSuccessAt,
-          lastFailureMessage: `Signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}" has a time gap of ${gap.humanize()} (${gap.asMilliseconds()}ms), and could be missing signals within that time. Consider increasing your look behind time or adding more Kibana instances.`,
-          lastSuccessMessage: currentStatusSavedObject.attributes.lastSuccessMessage,
-        });
 
-        if (ruleStatusSavedObjects.saved_objects.length >= 6) {
-          // delete fifth status and prepare to insert a newer one.
-          const toDelete = ruleStatusSavedObjects.saved_objects.slice(5);
-          await toDelete.forEach(async item =>
-            services.savedObjectsClient.delete(ruleStatusSavedObjectType, item.id)
-          );
-        }
-      }
+      const gap = getGapBetweenRuns({ previousStartedAt, interval, from, to });
+
+      writeGapErrorToSavedObject({
+        alertId,
+        logger,
+        ruleId: ruleId ?? '(unknown rule id)',
+        currentStatusSavedObject,
+        services,
+        gap,
+        ruleStatusSavedObjects,
+        name,
+      });
       // set searchAfter page size to be the lesser of default page size or maxSignals.
       const searchAfterSize =
         DEFAULT_SEARCH_AFTER_PAGE_SIZE <= params.maxSignals
```

Comment:
Also possible opportunity for improvement here (suggestion from a call I had with @rylnd): 

could we maybe rewrite this to use the lodash/fp `min` function like so...

```node.js
const searchAfterSize = min(DEFAULT_SEARCH_AFTER_PAGE_SIZE, params.maxSignals);
```

Reply:
No worries, I've already got this conflicting change on my branch 😉